crates/llm-base/src/loader.rs

```diff
@@ -364,7 +376,29 @@ pub fn load<M: KnownModel>(
     })?;
     let mut reader = BufReader::new(&file);
 
-    let mut loader = Loader::new(load_progress_callback);
+    let tokenizer = if let Some(path) = vocab_path {
+        let tok = if !path.exists() && path.to_str().unwrap().matches("/").count() == 1 {
```

Comment:
So the difference between `my_repo/` and `my_repo/tokenizer.json`?

Reply:
If there only one "/" in the path and that there is no file or folder at the provided path we can assume that the path could be the repo name (ex: RWKV/rwkv-raven-14b) I didn't saw any repo on hf that had "/" in the name, i could be wrong in my implementation but i though it would be a good way to avoid  trying to download something if the user made a typo in the path

Reply:
Oh shit, I didn't realise that would actually *download* the tokenizer from the repo.

Okay, I'd suggest changing `vocab_path` to `vocabulary_source: VocabularySource`, where `VocabularySource` is an enum with `ModelEmbedded` (which will use the GGML vocabulary if it's available, and error if it's not), `TokenizerFile(PathBuf)`, and `TokenizerHfPretrained(String)`. The CLI can then have options for these, too, and the behaviour should be obvious.